tsdb/db.go

```diff
@@ -552,7 +552,7 @@ func validateOpts(opts *Options, rngs []int64) (*Options, []int64) {
 	return opts, rngs
 }
 
-func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs []int64) (db *DB, returnedErr error) {
+func open(dir string, l log.Logger, r prometheus.Registerer, opts *Options, rngs []int64) (_ *DB, returnedErr error) {
```

Comment:
Mixing name and unnamed returns like this is weird.

Making the error an explicit local variable would be cleaner, there's only one naked return here.

Reply:
A local variable could be error prone, we use `if err := ...` in many places and its easy to not set the local variable properly before returning. Binding it to the return variable seems more robust.

Reply:
You don't have to call the variable `err`.

Reply:
I tried having a local variable, but it adds additional lines to every error return where we are wrapping. I prefer keeping it binded to return variable to prevent any regression.

Reply:
That's not how it's coded though. Sometimes it's returning returnedError, other times it's returning err. This is inconsistent for no reason I can see. I'd suggest removing returnedError everywhere outside of the defer, as it's only causing confusion.

Reply:
Now I see! I have removed returnedErr outside the defer now

Reply:
Thanks